src/ansys/mapdl/core/mapdl_grpc.py

```diff
@@ -2149,87 +2166,173 @@ def download(
+        for file in list_files:
+            # file is a complete path
+            basename = os.path.basename(file)
+            if os.path.isfile(os.path.join(target_dir, basename)):
+                os.remove(os.path.join(target_dir, basename))
+                # the file might have been already downloaded.
+                warn(
+                    f"The file {file} has been updated in the current working directory."
+                )
+
+            if os.path.isdir(os.path.join(self.directory, file)):
+                copy_files_to_the_root(
+                    os.path.join(self.directory, file), target_dir, recursive=recursive
+                )
+
+            else:
+                shutil.copy(
+                    os.path.join(self.directory, file),
+                    os.path.join(target_dir, basename),
+                )
+
+        return list_files
+
+    def _download_from_remote(
+        self,
+        files,
+        target_dir=None,
+        extension=None,
+        chunk_size=None,
+        progress_bar=None,
+    ):
+        """Download files when we are connected to a remote session."""
+
+        if isinstance(files, str):
+            list_files = self._validate_files(files, extension=extension)
+
+        elif isinstance(files, list):
+            if not all([isinstance(each, str) for each in files]):
+                raise ValueError(
+                    "The parameter `'files'` can be a list or tuple, but it should only contain strings."
+                )
+            list_files = []
+            for each in files:
+                list_files.extend(self._validate_files(each, extension=extension))
+
         else:
-            target_dir = os.getcwd()
+            raise ValueError(
+                f"The `file` parameter type ({type(files)}) is not supported."
+                "Only strings, tuple of strings or list of strings are allowed."
+            )
 
         for each_file in list_files:
-            try:
-                file_name = os.path.basename(
-                    each_file
-                )  # Getting only the name of the file.
-                #  We try to avoid that when the full path is supplied, it will crash when trying
-                # to do `os.path.join(target_dir"os.getcwd()", file_name "full filename path"`
-                # This will produce the file structure to flat out, but it is find, because recursive
-                # does not work in remote.
-                self._download(
-                    each_file,
-                    out_file_name=os.path.join(target_dir, file_name),
-                    chunk_size=chunk_size,
-                    progress_bar=progress_bar,
+            self._download(
+                each_file,
+                out_file_name=os.path.join(target_dir, each_file),
+                chunk_size=chunk_size,
+                progress_bar=progress_bar,
+            )
+
+        return list_files
+
+    def _validate_files(self, file, extension=None):
```

Comment:
I think this function is over-complicated. I'm having a look on my local branch.